modules/cnf-understanding-templates.adoc

```diff
@@ -0,0 +1,106 @@
+* Node-level templates that can reference both cluster-specific and node-specific fields.
+
+You can customize the templated fields as the {sco} supports all link:https://masterminds.github.io/sprig/[sprig library functions].
+Additionally, it provides two new functions that you can use while creating your custom manifests:
+
+`toYaml`:: The `toYaml` function encodes an item into a YAML string. If the item cannot be converted to YAML the function returns an empty string.
+
+`anyFieldDefined`:: The `anyFieldDefined` function checks if a specific field is defined. If defined, the {sco} renders the field as instructed.
+
+.Using .anyFieldDefined in the `ClusterInstance.Spec.Proxy` field
+[source,yaml]
+----
+[...]
+{{ if (anyFieldDefined .Spec.Proxy) }}
+    proxy:
+{{ .Spec.Proxy | toYaml | indent 4 }}
+{{ end }}
+[...]
+----
+
+The {sco} provides the following default, validated, immutable set of templates:
+
+.Default set of templates
+[cols=3*, width="95%", options="header"]
+|====
+
+|Installation method
+|Template type
+|Template content
+
+.2+|Assisted Installer
+a|Cluster-level templates +
+(`ai-cluster-templates-v1.yaml`)
+a|`ClusterDeployment` +
+`InfraEnv` +
+`KlusterletAddonConfig` +
+`ManagedCluster` +
+`AgentClusterInstall`
+
+a|Node-level templates +
+(`ai-node-templates-v1.yaml`)
+a|`NMStateConfig` +
+`BareMetalHost`
+
+.2+|Image-based Install Operator
+a|Cluster-level templates +
+(`ibi-cluster-templates-v1.yaml`)
+a|`ClusterDeployment` +
+`KlusterletAddonConfig` +
+`ManagedCluster` +
+
+a|Node-level templates +
+(`ibi-node-templates-v1.yaml`)
+a|`ImageClusterInstall` +
+`NetworkConfigMap` +
+`BareMetalHost` +
+
+|====
+
+[id="cnf-special-template-variable_{context}"]
+== Special templates variables
+
+The {sco} provides a set of special template variables that you can use in your templates.
+
+`CurrentNode`:: The {sco} explicitly controls the iteration of the node objects and exposes this variable to access all the content for the current node being handled in templating.
+`InstallConfigOverrides`:: This variable contains the merged `networkType`, `cpuPartitioningMode` and `installConfigOverrides` content.
+`ControlPlaneAgents`:: This variable is automatically derived from the `ClusterInstance` node objects. It consists of the number of control plane agents.
+`WorkerAgents`:: This variable is automatically derived from the `ClusterInstance` node objects. It consists of the number of worker agents.
+
+You can create a custom templated field by capitalizing the field name in the text template.
+For example, if you want to specify the `name` and `namespace` for your current node by using the `CurrentNode` special template variable, use the field names in the following form:
+
+[source,yaml]
+----
+name: "{{ .SpecialVars.CurrentNode.HostName }}"
+namespace: "{{ .Spec.ClusterName }}"
+----
+
+[id="cnf-sync-wave-annotation_{context}"]
+== Using the `siteconfig.open-cluster-management.io/sync-wave` annotation
```

Comment:
Should the heading here outline what this section is about, rather than how it's done. Perhaps something like, "Controlling the manifest order" or something like that